pkg/errctx/center.go

```diff
@@ -0,0 +1,40 @@
+package errctx
+
+import (
+	"context"
+
+	"go.uber.org/atomic"
+)
+
+type ErrCenter struct {
+	hasErr atomic.Bool
+	errVal atomic.Error
+	doneCh chan struct{}
+}
+
+func NewErrCenter() *ErrCenter {
+	return &ErrCenter{
+		doneCh: make(chan struct{}),
+	}
+}
+
+func (c *ErrCenter) OnError(err error) {
+	if err == nil {
+		return
+	}
+	if c.hasErr.Swap(true) {
+		// OnError is no-op after the first call with
+		// a non-nil error.
+		return
+	}
```

Comment:
If some of scenarios need to record the duplicated errors, should we provide an option to do that.

Reply:
Allowing duplicate errors can make this utility too difficult to use. I added a log instead.